File: src/main/java/nl/deltadak/plep/ui/taskcell/TaskConverter.java
package nl.deltadak.plep.ui.taskcell;

import javafx.scene.control.TreeCell;
import javafx.util.StringConverter;
import nl.deltadak.plep.HomeworkTask;

/**
 * custom stringconverter to define what editing a listcell means
 * this converter is set on each listcell
 */
public class TaskConverter extends StringConverter<HomeworkTask> {
    
    private final TreeCell<HomeworkTask> cell;
    
    TaskConverter(final TreeCell<HomeworkTask> cell) {
        this.cell = cell;
    }
    
    @Override
    public String toString(final HomeworkTask homeworkTask) {
        return homeworkTask.getText();
    }
    
    @Override
    public HomeworkTask fromString(final String string) {
        HomeworkTask homeworkTask = cell.getItem();
        homeworkTask.setText(string);
        
        return homeworkTask;
    }
}

File: src/main/kotlin/nl/deltadak/plep/ui/taskcell/treecell/TaskConverter.kt
package nl.deltadak.plep.ui.taskcell.treecell

import javafx.scene.control.TreeCell
import javafx.util.StringConverter
import nl.deltadak.plep.HomeworkTask

/**
 * Custom stringconverter to define what editing a [TreeCell] means.
 * This converter is for one TreeCell.
 */
class TaskConverter(
        /** TreeCell for which the converter is. */
        private val cell: TreeCell<HomeworkTask>) : StringConverter<HomeworkTask>() {

    /**
     * Converts the HomeworkTask provided into its string form.
     */
    override fun toString(homeworkTask: HomeworkTask?): String = homeworkTask?.text ?: ""

    /**
     * Converts a string into a HomeworkTask.
     */
    override fun fromString(string: String): HomeworkTask = cell.item.apply { text = string }

}
